extensions/sockets/src/main/java/org/mule/module/socket/api/exceptions/MisconfiguredSocketException.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.exceptions;
+
+public class MisconfiguredSocketException extends RuntimeException
```

Comment:
what's the purpose of this exception. How do you differenciate between a permanently illegal configuration (e.g: listening on port 1) vs. something that you might recover from (a port temporarily taken which might free up after some retries).

Shoudn't you just use a ConnectionException?